exp/controllers/ocimachinepool_controller.go

```diff
@@ -0,0 +1,602 @@
+			metadata = make(map[string]string)
+		}
+		metadata["user_data"] = base64.StdEncoding.EncodeToString([]byte(cloudInitData))
+
+		subnetId := machinePoolScope.GetWorkerMachineSubnet()
+		nsgId := machinePoolScope.GetWorkerMachineNSG()
+
+		launchInstanceDetails := core.ComputeInstanceDetails{
+			LaunchDetails: &core.InstanceConfigurationLaunchInstanceDetails{
+				CompartmentId: common.String(machinePoolScope.OCICluster.Spec.CompartmentId),
+				DisplayName:   common.String(machinePoolScope.OCIMachinePool.GetName()),
+				Shape:         common.String(machinePoolScope.OCIMachinePool.Spec.InstanceConfiguration.InstanceDetails.Shape),
+				SourceDetails: core.InstanceConfigurationInstanceSourceViaImageDetails{
+					ImageId: common.String(machinePoolScope.OCIMachinePool.Spec.ImageId),
+				},
+				Metadata: metadata,
+				CreateVnicDetails: &core.InstanceConfigurationCreateVnicDetails{
+					SubnetId:       subnetId,
+					AssignPublicIp: common.Bool(true),
+					NsgIds:         []string{*nsgId},
+				},
+			},
+		}
+
+		shapeConfig, err := machinePoolScope.BuildInstanceConfigurationShapeConfig()
+		if err != nil {
+			conditions.MarkFalse(machinePoolScope.MachinePool, infrav1exp.LaunchTemplateReadyCondition, infrav1exp.LaunchTemplateCreateFailedReason, clusterv1.ConditionSeverityError, err.Error())
+			machinePoolScope.Info("failed to create instance configuration due to shape config")
+			return err
+		}
+		if (shapeConfig != core.InstanceConfigurationLaunchInstanceShapeConfigDetails{}) {
+			launchInstanceDetails.LaunchDetails.ShapeConfig = &shapeConfig
+		}
+
+		req := core.CreateInstanceConfigurationRequest{
+			CreateInstanceConfiguration: core.CreateInstanceConfigurationDetails{
+				CompartmentId:   common.String(machinePoolScope.OCICluster.Spec.CompartmentId),
+				DisplayName:     common.String(machinePoolScope.OCIMachinePool.GetName()),
+				FreeformTags:    tags,
+				InstanceDetails: launchInstanceDetails,
+			},
+		}
+
+		resp, err := machinePoolScope.ComputeManagementClient.CreateInstanceConfiguration(ctx, req)
+		if err != nil {
+			conditions.MarkFalse(machinePoolScope.MachinePool, infrav1exp.LaunchTemplateReadyCondition, infrav1exp.LaunchTemplateCreateFailedReason, clusterv1.ConditionSeverityError, err.Error())
+			machinePoolScope.Info("failed to create instance configuration")
+			return err
+		}
+
+		machinePoolScope.SetInstanceConfigurationIdStatus(*resp.Id)
+		return machinePoolScope.PatchObject(ctx)
+	}
+
+	return nil
+}
+
+func (r *OCIMachinePoolReconciler) findInstancePool(ctx context.Context, machinePoolScope *scope.MachinePoolScope) (*core.InstancePool, error) {
+	// We have to first list the pools to get the instance pool.
+	// List returns InstancePoolSummary which lacks some details of InstancePool
+
+	reqList := core.ListInstancePoolsRequest{
+		CompartmentId: common.String(machinePoolScope.OCICluster.Spec.CompartmentId),
+		DisplayName:   common.String(machinePoolScope.OCIMachinePool.GetName()),
```

Comment:
do we set the name, sorry just want to put it out there so that I dont miss it

Reply:
It is set in the template 
```
  name: ${CLUSTER_NAME}-mp-0
```